slime_environments/environments/SlimeEnvMultiAgent.py

```diff
@@ -577,15 +614,28 @@ def reward_cluster_and_time_punish_time(self, current_agent):  # DOC NetLogo rew
                      (cluster / self.cluster_threshold) * (self.reward ** 2) + \
                      (((self.episode_ticks - self.cluster_ticks[self.agent]) / self.episode_ticks) * self.penalty)
 
-        self.rewards[self.agent].append(cur_reward)
+        self.rewards_cust[self.agent].append(cur_reward)
         return cur_reward
 
-    def reset(self):
+    def reset(self, seed=None, return_info=True, options=None):
         # empty stuff
         pop_tot = self.population + self.learner_population
-        self.rewards = {i: [] for i in range(self.population, pop_tot)}
+        self.rewards_cust = {i: [] for i in range(self.population, pop_tot)}
         self.cluster_ticks = {i: 0 for i in range(self.population, pop_tot)}
-        self.obs_dict = {a: BooleanSpace(size=2) for a in self.agents}
+        
+        #Initialize attributes for PettingZoo Env
+        self.agents = self.possible_agents[:]
+        self._agent_selector.reinit(self.agents)

```

Comment:
what is the difference between reinit (used here) and reset (used far above)?

Reply:
![image](https://github.com/smarianimore/RL-swarms/assets/86671201/c9368059-ab68-45ce-bee1-745a900a00a7)

These are definitions from pettingZoo's github. They both almost does the same thing: reset returns the current agent, whereas reinit just reorders. 
Do you mean the reset() in __init__ method? (line 168). If so, that reset is called whenever the env is first created, whereas the reinit here(line 628) reorders everytime an episode starts.

Reply:
many thanks for the pointer to the documentation :)
as far as I understood it is just a way to decouple ordering and selection of first agent, in fact reset calls reinit in turn.
Fine as it is, thanks.